plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/QueryExecutionContextAppender.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.spi.connector.ConnectorSession;
+
+public interface QueryExecutionContextAppender
```

Comment:
RemoteQueryModifier?

Reply:
I'm not sure I like this name, but I'm open to suggestions. @skrzypo987 @Praveen2112 For me it's only about appending query context (so additional information about the query execution) and in this particular case as a comment. Maybe I should remove the interface and leave just the name of the actual implementation, as there won't be any other implementations of this class where we are enabling this feature for every sql based connector.